### ISSUE TITLE
[Feature request] Add support for async programming

### ISSUE THREAD

**Issue by MohoWu** [rstudio/dygraphs#202]
With the release of Shiny 1.1.0 officially supporting async programming, may I ask is there any plan in the near future to make `dygraphOutput` and `renderDygraph` promise-aware?

**Comment by jjallaire** [comment]
@jcheng5 Will this come automatically with https://github.com/ramnathv/htmlwidgets/pull/316 ?

**Comment by jcheng5** [comment]
@jjallaire Yes, that's correct! Any widgets that use the typical htmlwidgets::shinyRenderWidget impl get async for free.
